src/core/shape_inference/include/sparse_fill_empty_rows_shape_inference.hpp

```diff
@@ -0,0 +1,114 @@
+                                 const std::vector<TShape>& input_shapes,
+                                 const ITensorAccessor& tensor_accessor = make_tensor_accessor()) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() == 4);
+
+    const auto& values_shape = input_shapes[0];
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           values_shape.rank().compatible(1),
+                           "The values input must be a 1D tensor.",
+                           values_shape);
+
+    const auto& dense_shape = input_shapes[1];
+    const bool is_dense_shape_rank_dynamic = dense_shape.rank().is_dynamic();
+    const bool is_dense_shape_valid =
+        is_dense_shape_rank_dynamic || (dense_shape.size() == 1 && dense_shape[0].compatible(2));
+    NODE_SHAPE_INFER_CHECK(
+        op,
+        input_shapes,
+        is_dense_shape_valid,
+        "The dense_shape input must be 1D and have exactly 2 elements, meaning only 2D sparse tensors are supported.");
+
+    const auto& indices_shape = input_shapes[2];
+    const bool is_indices_shape_valid = indices_shape.rank().is_dynamic() ||
+                                        (indices_shape.size() == 2 && indices_shape[1].compatible(2) &&
+                                         (is_dense_shape_rank_dynamic || indices_shape[0].compatible(values_shape[0])));
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           is_indices_shape_valid,
+                           "The indices input must be a 2D tensor with the first dimension matching the size of values "
+                           "and the second dimension having 2 elements.",
+                           indices_shape);
+
+    const auto& default_value_shape = input_shapes[3];
+    NODE_SHAPE_INFER_CHECK(op,
+                           input_shapes,
+                           default_value_shape.rank().compatible(0),
+                           "The default_value input must be a scalar.",
+                           default_value_shape);
+
+    auto output_shapes = std::vector<TRShape>(3);
+    auto& output_indices_shape = output_shapes[0];
+    auto& output_values_shape = output_shapes[1];
+    auto& empty_row_indicator_shape = output_shapes[2];
+    output_indices_shape.resize(2);
+    output_values_shape.resize(1);
+    empty_row_indicator_shape.resize(1);
+    output_indices_shape[1] = 2;  // Only 2D cases are supported
+
+    if (auto dense_shape_value = get_input_const_data_as_shape<TRShape>(op, 1, tensor_accessor)) {
+        const auto& number_of_rows = (*dense_shape_value)[0].get_length();
+        empty_row_indicator_shape[0] = number_of_rows;
+
+        if (auto indices_value = get_input_const_data_as<TRShape, int64_t>(op, 2, tensor_accessor)) {
+            auto is_valid_index = [](int64_t index, int64_t max_value) -> bool {
+                return index >= 0 && index < max_value;
+            };
```

Comment:
It will throw for negative indices, probably something worth to mention in the op specification.

Reply:
I added it, but it's also worth to remember in the future in case we have more sparse operators to add some sort of unified spec for a sparse tensor representation in OpenVINO.